Gem/CMakeLists.txt

```diff
@@ -30,15 +30,15 @@ if(NOT PAL_TRAIT_ROBOT_VACUUM_SAMPLE_SUPPORTED)
     return()
 endif()
 
-# We are on a supported platform, so add the ROS2-Gem-Demo target
-# Note: We include the common files and the platform specific files which are set in ros2-gem-demo_files.cmake and
-# in ${pal_dir}/ros2-gem-demo_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
+# We are on a supported platform, so add the RobotVacuumSample target
+# Note: We include the common files and the platform specific files which are set in robot-vacuum-sample_files.cmake and
+# in ${pal_dir}/robot_vacuum_sample_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
 ly_add_target(
-    NAME ROS2-Gem-Demo.Static STATIC
+    NAME RobotVacuumSample.Static STATIC
```

Comment:
Can you update the `prefix` for the CMake targets to use the [${gem_name}](https://github.com/o3de/o3de/blob/development/Templates/DefaultProject/Template/Gem/CMakeLists.txt#L51) variable, that would allow renaming the project gem by only updating the gem.json file?